### DISCUSSION TITLE
Upload Not Completing

### DISCUSSION THREAD

**Discussion by ddlidded** [requarks/wiki#4830] [Error / Bug Report]
When uploading an image-based asset through the browse -> upload method, it gets to 100% but does not complete and does not get uploaded successfully.

When checking my application logs, this is a message that is displayed:
warn: Cannot read properties of undefined (reading 'toLowerCase')

If it helps, this issue was not occurring a day or two ago and was working fine.

Thanks in advance guys!

**Comment by ddlidded** [comment]
Now I am receiving the message "Error during upload"

**Reply by sanjibm** [reply]
Facing the exact same issue. 

There must be an easy way to enable more detailed logging to see whats going on.
Using docker compose following installation instructions from: 
https://docs.requarks.io/install/docker

Using this docker compose file:
```
version: "3"
services:

  db:
    image: postgres:11-alpine
    environment:
      POSTGRES_DB: wiki
      POSTGRES_PASSWORD: wikijsrocks
      POSTGRES_USER: wikijs
    logging:
      driver: "none"
    restart: unless-stopped
    volumes:
      - db-data:/var/lib/postgresql/data

  wiki:
    image: requarks/wiki:1.8
    depends_on:
      - db
    environment:
      DB_TYPE: postgres
      DB_HOST: db
      DB_PORT: 5432
      DB_USER: wikijs
      DB_PASS: wikijsrocks
      DB_NAME: wiki
    restart: unless-stopped
    ports:
      - "8085:3000"
    volumes:
      - /share/Document/wiki/config:/config
      - /share/Document/wiki/data:/data
      - /share/Document/wiki/backups:/backups

volumes:
  db-data:
```

**Comment by visionlabgit** [comment]
> warn: Cannot read properties of undefined (reading 'toLowerCase')

I also encountered the same issue (with the same 'toLowerCase' warning in logs), which may have been solved now.

I installed wiki.js via docker a few days ago on a remote ubuntu. Everything worked fine except for this asset uploading failure.
Suspecting a reverse proxy config issue, I installed a new instance via docker today on my local mac desktop. Asset uploading was OK. Noticing this issue was opened just a few days ago, and @ddlidded said it used to be OK before, I checked the docker image version. Indeed, the version was changing fast recently.
On the ubuntu,
```
REPOSITORY                       TAG       IMAGE ID       CREATED       SIZE
requarks/wiki                    2         7573dcfed162   6 days ago    441MB
```
On the mac,
```
REPOSITORY        TAG       IMAGE ID       CREATED         SIZE
requarks/wiki     2         e6969a016b78   4 days ago      441MB
```
So I reinstalled wiki.js on the ubuntu using the latest docker image, and the problem was fixed.

In short, this could be a bug accidentally introduced in version 7573dcfed162, and quickly fixed in the subsequent version.

**Reply by NGPixel** [reply]
Correct, upgrade to the latest version to fix this issue.
